fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -1537,6 +1550,11 @@ public static IndexState writeOnlyIfTooManyRecordsForRebuild(long recordCount, b
             // on an unset boolean field results in getting back "false".)
             omitUnsplitRecordSuffix = info.getOmitUnsplitRecordSuffix();
         }
+        if (indexStateIsStoredInOldFormat(storeHeader)) {
```

Comment:
I'm not sure if I follow this. Would you please elaborate?

Reply:
Right now (unless I'm missing something), the update to store the data in the new location does not check to see if the format version is greater than or equal to the new format version, so if someone creates a record store object at an older format version, the data will get converted over here, so the record store will functionally be in the new format, but it will still claim to be in the old format (based on the store header). I think.

Reply:
Ok - for this one too, I spoke with Alec offline. I misunderstood what `checkUserVersion` does. no need to move the upgrade to after `checkUserVersion`. This is done now.